core/templates/dev/head/services/StateTopAnswersStatsService.js

```diff
@@ -83,14 +83,16 @@ oppia.factory('StateTopAnswersStatsService', [
 
     var onStateRenamed = function(oldStateName, newStateName) {
       onStateAdded(newStateName);
-      // Swap the values before deleting.
-      var cache = stateTopAnswersStatsCache;
-      cache[newStateName] =
-        [cache[oldStateName], (cache[oldStateName] = cache[newStateName])][0];
+      // Swap so onStateDeleted is allowed to assume *anything* in the cache of
```

Comment:
I don't understand why we're modelling a rename as an add + a delete. There is a subtle difference: in a rename, the identity of the state is preserved, whereas with an add + delete, it isn't.

I still think the approach that you had a couple of iterations ago, where you just moved the value to a different key, is much cleaner. Why did we move away from that?

Reply:
Hmm, I moved away b/c I figured it was easier to manage if we have it in terms of what needs to happen (need to add a new state, need to delete old state name, need to transfer ownership).

I also wanted to do the swapping to simplify future code, but I see your point. Also, I figure that once state ID is in, rename will be useless anyway, so I'll just go back to the old iteration!